### PR TITLE
Switch Navigation to first Car profile when connected to CarPlay

### PR DESCRIPTION
**Pull request by DmitrySvetlichny**

### CODE REVIEW

File: Sources/OAApplicationMode.h [hunk 1]
```diff
@@ -74,6 +74,7 @@
 - (BOOL) hasFastSpeed;
 - (int) getRouteTypeProfile;
 - (BOOL) isDerivedRoutingFrom:(OAApplicationMode *)mode;
++ (BOOL) isAppModeDerivedFromCar:(OAApplicationMode *)mode;
```
Comment (tigrim) [inline Sources/OAApplicationMode.h:77]: ?

File: Sources/AppHost/CarPlayDelegate/CarPlaySceneDelegates.swift [hunk 2]
```diff
@@ -81,6 +77,51 @@ final class CarPlaySceneDelegate: UIResponder {
...
+    
+    private func configureStandardSettings() {
+        guard let settings = OAAppSettings.sharedManager() else { return }
+        let lastUsedMode = settings.useLastApplicationModeByDefault.get() ? OAApplicationMode.value(ofStringKey: settings.lastUsedApplicationMode.get(), def: OAApplicationMode.default()) : settings.defaultApplicationMode.get()
+        settings.setApplicationModePref(lastUsedMode, markAsLastUsed: false)
+        OARoutingHelper.sharedInstance().setAppMode(lastUsedMode)
+    }
+    
+    private func findAppropriateCarMode(defaultAppMode: OAApplicationMode) -> OAApplicationMode? {
+        if !OAApplicationMode.isAppModeDerived(fromCar: defaultAppMode) {
```
Comment (tigrim) [inline Sources/AppHost/CarPlayDelegate/CarPlaySceneDelegates.swift:115]: defaultAppMode.isDerivedRouting(from: OAApplicationMode.car()) ?

File: Sources/AppHost/CarPlayDelegate/CarPlaySceneDelegates.swift [hunk 3]
```diff
@@ -81,6 +77,51 @@ final class CarPlaySceneDelegate: UIResponder {
...
+    
+    private func configureCarPlaySettings(initialAppMode: OAApplicationMode) {
+        guard let settings = OAAppSettings.sharedManager(), let routingHelper = OARoutingHelper.sharedInstance() else { return }
+        if settings.isCarPlayModeDefault.get() {
+            let carPlayMode = findAppropriateCarMode(defaultAppMode: initialAppMode)
+            settings.setApplicationModePref(carPlayMode, markAsLastUsed: false)
+            routingHelper.setAppMode(carPlayMode)
+        } else {
+            let carPlayMode = settings.carPlayMode.get()
+            settings.setApplicationModePref(carPlayMode, markAsLastUsed: false)
```
Comment (tigrim) [inline Sources/AppHost/CarPlayDelegate/CarPlaySceneDelegates.swift:102]: Code duplication, move it outside the if statement

**Comment by alex-osm** [comment]
Make code more clear (similar to java) - https://github.com/osmandapp/OsmAnd/blob/a79f67f02273645e4c9dde2c07b655b3ee504d6e/OsmAnd/src/net/osmand/plus/auto/NavigationSession.java#L188

**Pull request merged**